zipline/pipeline/graph.py

```diff
@@ -190,8 +190,23 @@ def _add_to_graph(self, term, parents, extra_rows):
         # Number of extra rows we need to compute for this term's dependencies.
         dependency_extra_rows = extra_rows + term.extra_input_rows
 
+        if isinstance(term, ComputableTerm):
+            # For computable terms, we want to manually add the term's mask to
+            # the graph with zero extra rows. A computable term does not
+            # directly require its mask to have any extra rows. Only loadable
+            # terms should dictate how many extra rows a mask should compute.
+            self._add_to_graph(
+                term.mask,
+                parents,
+                extra_rows=0,
```

Comment:
I think this actually wants to be `extra_rows` here, not 0.  If we need 3 extra rows of a `ComputableTerm`, then we also need 3 extra rows of the mask.  What we **don't** need is an additional `term.extra_input_rows` of the mask.